lib/train/transports/local.rb

```diff
@@ -160,12 +174,22 @@ def run_command(cmd)
           Local::CommandResult.new(res.stdout, res.stderr, res.exitstatus)
         end
 
+        def close
+          Process.kill("KILL", @server_pid)
+          @server_pid = nil
+        end
+
         private
 
         def acquire_pipe
+          require "win32/process"
+
           pipe_name = "inspec_#{SecureRandom.hex}"
 
-          start_pipe_server(pipe_name)
+          @server_pid = start_pipe_server(pipe_name)
+
+          # Ensure process is killed when the Train process exits
+          at_exit { Process.kill("KILL", @server_pid) unless @server_pid.nil? }
```

Comment:
my ruby is rusty but could you call `self.close` instead of repeating the `Process.kill` here

Reply:
Thanks @mattray.  Did the required changes.

Reply:
I hit that exact line of code yesterday for another reason. Maybe we can include the fix to that here as well?

When programmatically using InSpec from a custom Chef resource, the broken pipe of the KILL signal results in a failed Chef run as the `at_exit` will make the run exit non-zero on exceptions. I had a simple `rescue true` suffix for the `Process.kill` command at hand.

Question only remains if this PR should be merged first and I hand in my addition of if we append the fix here.

Reply:
At this point, we're only killing the subprocess if we know the PID - that is, we believe that the process is running. So I think that should cover the common cases. I think absorbing all exceptions will likely hide new problems. I'm reluctant to add a catch-all rescue unless we must.